tools/pubsys/src/aws/validate_ami/mod.rs

```diff
@@ -0,0 +1,799 @@
+//! The validate_ami module owns the 'validate-ami' subcommand and controls the process of validating
+//! EC2 images
+
+pub(crate) mod ami;
+pub(crate) mod results;
+
+use self::ami::ImageDef;
+use self::results::{AmiValidationResult, AmiValidationResultStatus, AmiValidationResults};
+use crate::aws::client::build_client_config;
+use crate::aws::validate_ami::ami::describe_images;
+use crate::Args;
+use aws_sdk_ec2::{Client as AmiClient, Region};
+use log::{info, trace};
+use pubsys_config::InfraConfig;
+use snafu::ResultExt;
+use std::collections::{HashMap, HashSet};
+use std::fs::File;
+use std::path::PathBuf;
+use structopt::{clap, StructOpt};
+
+/// Validates EC2 images
+#[derive(Debug, StructOpt)]
+#[structopt(setting = clap::AppSettings::DeriveDisplayOrder)]
+pub(crate) struct ValidateAmiArgs {
+    /// File holding the expected amis
+    #[structopt(long, parse(from_os_str))]
+    expected_amis_path: PathBuf,
+
+    /// Optional path where the validation results should be written
+    #[structopt(long, parse(from_os_str))]
+    write_results_path: Option<PathBuf>,
+
+    #[structopt(long, requires = "write-results-path")]
+    /// Optional filter to only write validation results with these statuses to the above path
```

Comment:
nit, but the help text looks funky without this.

```suggestion
    /// Optional filter to only write validation results with these statuses to the above path.
```